src/main/java/qupath/ext/align/gui/ImageServerOverlay.java

```diff
@@ -115,7 +126,21 @@ public void setRenderer(ImageRenderer renderer) {
 	public Affine getAffine() {
 		return affine;
 	}
-	
+
+	/**
+	 * Reset the affine transform to its pixel-correct scaled identity
+	 */
+	public void resetAffine() {
+		if (this.affine == null)
+			return;
+
+        // Calculate the a and y scaling factor
```

Comment:
I haven't tried it, but this looks like it wouldn't handle the case where the pixel size information is missing for at least one image.

Technically, the pixel width and height *might* not be in µm - you can use e.g. `PixelCalibration.getPixelWidthUnit()` to check that the values match for the two calibration objects, and then use `PixelCalibration.getPixelWidth()` to calculate the ratios *only* if they do.

Reply:
Right! I implemented a ~half ar...~ halfway solution in https://github.com/qupath/qupath-extension-align/pull/10/commits/d6e140068494361f3ae107a6fa6debebc942f2d1 which relies on `hasPixelSizeMicrons()` being true, so that none of `getPixelWidthMicrons()` and `getPixelHeightMicrons()` are NaN in my ratios.

I can totally check matching pixel units though if you prefer (like you asked).